Part 1
1.One must engage in a designed experiment to be able establish a cause and effect conclusion between the indepdent and dependent variable. Simply observing is not taking into the many confounding variables that may be contributing to a certain conclusion. For example Promotion 1 may have been chosen by stores with the most crowded areas, leading to a higher sale resulting from not the specific promotion, but rather due to the general number of customers.


2.The experimental unit are the individual stores, the treatment is the promotion number, and the outcome is the SalesInThousands categorized by type of promotion. 


3.This is primarily an inference problem as the company is trying to see what and if promotion is impacting sales, not attempting to make a prediction, which is a claim about the future using past data. 

Part 2

In [1]:
import pandas as pd 
import numpy as np
import plotly.express as px 


In [ ]:
#1  
data = pd.read_csv("WA_Marketing-Campaign.csv")
data["Promotion"] = data["Promotion"].astype(str)
fig = px.bar(
    data,
    x="LocationID",              
    y="SalesInThousands",
    color="Promotion",           
    barmode="group",             
    title="Sales by Store and Promotion"
)
fig.update_xaxes(type="category")   
fig.show()


2.At first glance, there are not a lot of obvious differences in typical sales. All promotion have stores that made above 300 (thounsands of dollars). Promotion 3 and 1 seems to have more stores that reach above the 200 mark. There is a lot of variation within each promotion as one can clearly see that each promotion has stores that had a lot more sales while there are stores that did not make as much across all three types. 


3.No because there hasn't been an analysis done yet to determine if there are meaningful differences that one can attribute to the differing promotions. Looking at the graph, one cannot determine this. 

Part 3
1.Theta is the population average dffierence in weekly sales between promotion 2 and 1. If theta is larger than 0, this means that promotions 2 on average made more weekly sales, if it is zero then that means both of the promotions on average made the same amount of sales, and if its less than 0, that means promotion 1 had on average more weekly sales than storees with promotion 2.

In [ ]:
#2
promGroup = data.groupby("Promotion")["SalesInThousands"].mean().reset_index()
prom1 = promGroup[promGroup["Promotion"] == "1" ]
prom2 = promGroup[promGroup["Promotion"] == "2" ]

theta = prom2["SalesInThousands"].iloc[0] - prom1["SalesInThousands"].iloc[0]
theta


,Promotion,SalesInThousands
0,1,58.099012


3.Stores using promotion 1 made around 10.77 thousands of dollars more weekly than stores using promotion 2.

In [4]:
#4
prom1_sales = (
    data.loc[
        data["Promotion"] == "1",
        "SalesInThousands"
    ]
    .to_numpy()
)

prom2_sales = (
    data.loc[
        data["Promotion"] == "2",
        "SalesInThousands"
    ]
    .to_numpy()
)
my_seed = np.random.default_rng(500)
B = 5000

boot_spend_diff = np.empty(B)

for b in range(B):

    promotion1 = my_seed.choice(
        prom1_sales,
        size=len(prom1_sales),
        replace=True
    )

    promotion2 = my_seed.choice(
        prom2_sales,
        size=len(prom2_sales),
        replace=True
    )

    boot_spend_diff[b] = (
        promotion2.mean()
        - promotion1.mean()
    )
lower, upper = np.percentile(
    boot_spend_diff,
    [2.5, 97.5]
)
print(lower, upper)

-14.03727044161306 -7.486401379267708


5.On average, promotion 1 had higher weekly sales than promotion 2 by around 7.49 thousands of dollars to 14.04 thousands of dollars. 


6.Because even if it contains 0, on average the difference is between the interval (which includes non zero numbers). Containing 0 just points to the fact that we cannot reject the null hypothesis meaning that the difference is not significant (the difference could have happened by variation/chance rather than due to the difference in promotion).

Part 4
1.No, the interval shows that on average promotion 1 had more weekly sales than promotion 2 by around 7.49 thousands of dollars to 14.04 thousands of dollars. 


2.No, the interval were negative numbers meaning promotion 1 sales were higher than promotion 2 sales by a significant amount as it did not include zero. 


3.Keep Promotion 1. The confidence interval excluded zero and its estimated effect is around on average 7.49 thousands of dollars to 14.04 thousands of dollars weekly more than promotion 2. 

Part 5 
1.On average, the amount of weekly sales between promotion 1,2,3 are the same. 


2.After ANOVA shows that there is a significant difference, one can decide which two to compare by referring back to the explanatory analysis.For example, there seems to be differnce between promotion 3 and 2, so I would do a boostrap condifence interval to comapare.


3.When there are seveal comparisons, the probability to see significant differences increase becasue one is doing more comparisons and thus increases the probability that one sees "significant" differences when it was actually simply due to chance. 

In [8]:
import statsmodels.api as sm
from statsmodels.formula.api import ols
model_sales = ols(
    "SalesInThousands ~ C(Promotion)",
    data=data
).fit()
anova_table = sm.stats.anova_lm(model_sales, typ=2)

anova_table

,sum_sq,df,F,PR(>F)
C(Promotion),11449.174291,2.0,21.953486,6.765849e-10
Residual,142114.105414,545.0,NaN,NaN


Part 6
1.Because now we are using past data to predict future unknown data rather than seeing what could explain the result of the original data. 


2.MarketSize, Promotion, and AgeOfStore


3.One would wait to see if the model's predictions are accurate by comparing it to future sales that it predicted. If the predictions are close to real data then one can say the model is successful. In this experiment, we are not trying to predict future data, we care about why we got that data and seeing if this one variable (promotions) led to differences in sales. 

AI Usage: 
I used AI and the class jupyter notebook to figure out how to correctly write the bootstrap/ANOVA code as well as producing the visualization that I wanted. I also used it to help debug errors when I was attempting to do calculations using the data from the csv or importing packages. 